### DISCUSSION TITLE
Tweaking the electrodeposition code to simulate one Atomic Layer Deposition cycle

### DISCUSSION THREAD

**Discussion by Rave169** [usnistgov/fipy#882] [General]
Hello All,

I am trying to simulate one Atomic Layer Deposition cycle using the diffusion approach and the level set method. I can see that in the earlier version of fipy, electrodeposition examples were introduced. I am thinking of two approaches:

1. I have currently tweaked the electrodeposition code to include the diffusion equation for catalyst, surfactant equation for surface coverage. I am getting the surface coverage across the surface but the boundary is not evolving since there is no term for velocity parameter.

2. In the second approach, I am including the metal diffusion equation and the metal boundary condition.

Can you please guide me which of the two approaches is more suitable to model ALD cycle. As the next step, I am thinking to simulate multiple ALD cycles.

![Untitled](https://user-images.githubusercontent.com/19670174/187915659-7b9f4e8b-c308-49a4-b99e-802352a8bd78.png)

**Comment by wd15** [comment]
Hi @Rave169, from the above, I'm a bit confused. Obviously, an expression for the velocity in terms of other variables in the system is required. Is it that you don't know how to derive this equation or is it that you have an equation, but that you don't know how to make it work with FiPy? Is the system that you have closed (variables vs equations)? Another way to ask this is can you solve the 1D steady state problem with paper and pencil?
